packages/plugins/users-permissions/server/strategies/users-permissions.js

```diff
@@ -20,24 +20,24 @@ const authenticate = async ctx => {
       const { id } = await getService('jwt').getToken(ctx);
 
       if (id === undefined) {
-        return { error: 'Invalid token: Token did not contain required fields' };
+        return { authenticated: false };
       }
 
       // fetch authenticated user
       const user = await getService('user').fetchAuthenticatedUser(id);
 
       if (!user) {
-        return { error: 'Invalid credentials' };
+        return { authenticated: false };
```

Comment:
I feel like  from here on this should actually throw as we have detected the token is a valid token for users &. permissions with the getToken then we know it won't match with another strategy.
It would be possible multiple strategies can parse the exact same token (e.g use the same jwt secret) but that would be a bad pattern. WDYT ?

Reply:
Good point 👌.

I'd say that if the token is valid but the data it relates to isn't, then we should definitely fail the authentication process. Especially when dealing with JWT. The way it's currently build they all have their own salt/secret. Thus they should never be a case that a token is validated in both strategies.

Reply:
My thoughts exactly. We can throw then :)